src/interpreter/built_ins.rs

```diff
@@ -47,7 +53,9 @@ mod chr {
         let i: i32 = interpreter.pop_integer();
         let mut s: String = String::new();
         s.push((i as u8) as char);
-        interpreter.function_result = s.into();
+        interpreter
+            .context_mut()
+            .set_variable(ResolvedDeclaredName::parse("CHR$"), s.into());
```

Comment:
Move the logic that generates the ResolvedDeclaredName into the enum builtInFunction